### ISSUE TITLE
Issue with Root Args and config file

### ISSUE THREAD

**Issue by efanibi25** [omni-us/jsonargparse#232]
## 🐛 Bug report
An error 
`jsonargparse.util.ParserError: Problem in default config file "/home/main/Tools/.config/config.yaml" :: Configuration check failed :: No action for destination key "prepare.output" to check its value.
`
is raised when trying to parse args

### To reproduce
```
parser = ArgumentParser(default_config_files=[os.path.join('.config/config.yaml')])
    parser.add_argument('-c','--config', action=ActionConfigFile,help="override config file")  
    parser.add_argument("-o","--output",help="Path to Dir",required=True)
    prepare = ArgumentParser(description="Prepare file to be uploaded")
    prepare.add_argument('-m','--media',help="Directory to retrive media files",required=True)
  subcommands = parser.add_subcommands()
    subcommands.add_subcommand('prepare',prepare, help="prepare mediafile for upload")

```
root argument must have required=True
root argument must also come before any group arguments. 

Also Note: I have been able to fix this by setting root args to be the last argument

It strange because it also can parses correctly if I set required=False

### Expected behavior
Parse arguments

### Environment

- jsonargparse version 4.19.0
- Python version 3.8
- How jsonargparse was installed (e.g. `pip install jsonargparse):
- OS Linux

**Comment by mauvilsa** [comment]
Thank you very much for reporting. I don't quite understand the description and tried to reproduce the problem but unsuccessfully. Can you please improve the reproduction section, such that:
- the code snippet is correct: all imports, valid indentation, inclusion of `parser.parse_args`
- how to run the code to get the error, i.e. `$ python cli.py ...`

What I tried to do was to run the code

```python
import os
from jsonargparse import ArgumentParser, ActionConfigFile

parser = ArgumentParser(default_config_files=[os.path.join('.config/config.yaml')])
parser.add_argument('-c','--config', action=ActionConfigFile,help="override config file")  
parser.add_argument("-o","--output",help="Path to Dir",required=True)
prepare = ArgumentParser(description="Prepare file to be uploaded")
prepare.add_argument('-m','--media',help="Directory to retrive media files",required=True)
subcommands = parser.add_subcommands()
subcommands.add_subcommand('prepare',prepare, help="prepare mediafile for upload")

print(parser.parse_args())
```
using as command
```bash
$ python cli.py -o . prepare --media value
Namespace(config=None, output='.', prepare=Namespace(media='value'), subcommand='prepare')
```
It executes without issue printing the namespace. Note that I ran it in a directory that only has the `cli.py` file and no configuration files. If the bug depends on there being config files, please also describe this.

**Comment by efanibi25** [comment]
Okay yeah here is a better description
Yes the bug does depend on the inclusion of a config file

filename:test.py

Code
```
import os
from jsonargparse import ArgumentParser, ActionConfigFile

parser = ArgumentParser(default_config_files=['.config/config.yaml'])
parser.add_argument('-c','--config', action=ActionConfigFile,help="override config file")  
parser.add_argument("-o","--output",help="Path to Dir",required=True)
prepare = ArgumentParser(description="Prepare file to be uploaded")
prepare.add_argument('-m','--media',help="Directory to retrive media files",required=True)
subcommands = parser.add_subcommands()
subcommands.add_subcommand('prepare',prepare, help="prepare mediafile for upload")

print(parser.parse_args())
```

configfile
path:config/config.yaml

```
output: test
prepare: 
    media: test
```

How to Trigger/Run
test.py prepare

**Comment by efanibi25** [comment]
As a note changing the code to this does work

import os
from jsonargparse import ArgumentParser, ActionConfigFile

parser = ArgumentParser(default_config_files=['.config/config.yaml'])
parser.add_argument('-c','--config', action=ActionConfigFile,help="override config file")  
prepare = ArgumentParser(description="Prepare file to be uploaded")
prepare.add_argument('-m','--media',help="Directory to retrive media files",required=True)
subcommands = parser.add_subcommands()
subcommands.add_subcommand('prepare',prepare, help="prepare mediafile for upload")
**parser.add_argument("-o","--output",help="Path to Dir",required=True)**


print(parser.parse_args())

**Comment by mauvilsa** [comment]
I created a pull request fixing the issue, see #233. Please review the changes and test the code. You can install directly from that branch by running:
```
pip install "jsonargparse @ https://github.com/omni-us/jsonargparse/zipball/issue-232"
```

**Comment by mauvilsa** [comment]
Note that the problem is `parser.add_subcommands()` so another workaround is to first call `add_subcommands` and then add arguments.
